src/gtil/predict.cc

```diff
@@ -187,6 +189,41 @@ struct MultiClfProbDistLeafOutputLogic {
   }
 };
 
+struct PredictLeafOutputLogic {
+  template <typename ThresholdType, typename LeafOutputType>
+  inline static void PushOutput(const treelite::Tree<ThresholdType, LeafOutputType>&,
+                                std::size_t, int node_id, float* output, std::size_t) {
+    *output = static_cast<float>(node_id);
```

Comment:
Am I interpreting this correctly as the overall id of the node within the tree as opposed to the index of this node among leaf nodes?

Reply:
Yes, we index nodes using all nodes in the tree, including leaf nodes and internal nodes.

Reply:
Okay. In that case, we probably need to be even more explicit about how the ID is defined in the documentation. How do other frameworks handle this? Is it always a breadth-first assignment with a consistent split direction (less than/contained in category list)?

Reply:
When reading tree models, Treelite re-assigns new IDs to nodes by traversing each tree breadth-first. So the root node gets ID 0, the left child of the root node gets 1, the right child of the root gets 2, and so forth.

Let me go ahead and add a comment about this behavior.